src/main/java/com/leinardi/pycharm/mypy/checker/Problem.java

```diff
@@ -52,17 +52,36 @@ public Problem(final PsiElement target,
         this.suppressErrors = suppressErrors;
     }
 
-    @NotNull
-    public ProblemDescriptor toProblemDescriptor(final InspectionManager inspectionManager) {
-        return inspectionManager.createProblemDescriptor(target,
-                MypyBundle.message("inspection.message", getMessage()),
-                null, problemHighlightType(), false, afterEndOfLine);
+    public void createAnnotation(@NotNull AnnotationHolder holder) {
+        String message = MypyBundle.message("inspection.message", getMessage());
+        AnnotationBuilder annotation = holder
+                .newAnnotation(getHighlightSeverity(), message)
+                .range(target.getTextRange());
+        if (isAfterEndOfLine()) {
+            annotation = annotation.afterEndOfLine();
+        }
+        annotation.create();
     }
 
     public SeverityLevel getSeverityLevel() {
         return severityLevel;
     }
 
+    @NotNull
+    public HighlightSeverity getHighlightSeverity() {
+        switch (severityLevel) {
+            case ERROR:
+                return HighlightSeverity.ERROR;
```

Comment:
I didn't realize this before, but I could switch all inspections to `HighlightSeverity.WARNING` to retain the previous style (yellow background).

![image](https://user-images.githubusercontent.com/137616/144712203-910102b9-2528-4ddb-a116-3941d36367d2.png)

But I've grown to like the red squiggly underline look. It's also familiar to users of IDEA TypeScript integration. I'd keep it `HighlightSeverity.ERROR` personally.

![image](https://user-images.githubusercontent.com/137616/144712229-a1cc5d5a-e7b4-4037-b389-8820b1d052a6.png)

Reply:
Ah I totally forgot that severities are configurable in the inspection profile. I suppose using that is preferable. We can revisit this logic in another PR.

![image](https://user-images.githubusercontent.com/137616/144712889-4d13a09f-ba4e-445d-8284-d2873dec619f.png)